src/messenger.rs

```diff
@@ -1,6 +1,8 @@
+#![feature(option_result_contains)]
+
 use super::keep_alive::{KeepAliveOperations, NewKeepAliveConnectionMessage};
 use super::packet::{write, Packet};
-use std::collections::HashMap;
+use std::collections::{HashMap, HashSet};
```

Comment:
using a hash set here for the broadcast lists, there's no reason to ever list a conn id twice and this ensures that never happens